src/plots/cartesian/set_convert.js

```diff
@@ -112,26 +113,23 @@ module.exports = function setConvert(ax) {
         if(ax.type === 'date') {
             // check if milliseconds or js date objects are provided for range
             // and convert to date strings
-            range[0] = Lib.cleanDate(range[0]);
-            range[1] = Lib.cleanDate(range[1]);
+            range[0] = Lib.cleanDate(range[0], BADNUM, calendar);
+            range[1] = Lib.cleanDate(range[1], BADNUM, calendar);
         }
 
         for(i = 0; i < 2; i++) {
             if(ax.type === 'date') {
-                if(!Lib.isDateTime(range[i])) {
+                if(!Lib.isDateTime(range[i], ax.calendar)) {
                     ax[rangeAttr] = dflt;
                     break;
                 }
 
-                if(range[i] < Lib.MIN_MS) range[i] = Lib.MIN_MS;
-                if(range[i] > Lib.MAX_MS) range[i] = Lib.MAX_MS;
```

Comment:
This was a relic (albeit not a very old one, as `MIN_MS` and `MAX_MS` only came in with my work a month or so back...) as `cleanDate` / `dateTime2ms` won't accept strings that give invalid ms output.